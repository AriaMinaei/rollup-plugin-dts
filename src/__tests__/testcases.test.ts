import * as rollup from "rollup";
import fs from "fs";
import path from "path";
import dts from "../";

const TESTCASES = path.join(__dirname, "testcases");

async function assertTestcase(dir: string) {
  const bundle = await rollup.rollup({
    input: path.join(dir, "index.ts"),
    plugins: [dts()],
  });

  const { code } = await bundle.generate({
    format: "es",
    // TODO: enable source maps at some pointâ€¦
    // sourcemap: true,
    // sourcemapExcludeSources: true,
  });

  expect(code.trim()).toMatchSnapshot();
}

describe("rollup-plugin-dts", () => {
  const dirs = fs.readdirSync(TESTCASES);
  for (let name of dirs) {
    const dir = path.join(TESTCASES, name);
    if (fs.statSync(dir).isDirectory()) {
      let testfn = it;
      if (name.startsWith("skip.")) {
        name = name.substr("skip.".length);
        testfn = it.skip;
      }
      testfn(`works for testcase "${name}"`, () => {
        return assertTestcase(dir);
      });
    }
  }
});
