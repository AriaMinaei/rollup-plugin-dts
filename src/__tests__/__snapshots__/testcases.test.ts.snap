// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rollup-plugin-dts works for testcase "export-simple-function" 1`] = `
"function foo()

export { foo };"
`;

exports[`rollup-plugin-dts works for testcase "export-simple-interface" 1`] = `
"interface Foo {}

export { Foo };"
`;

exports[`rollup-plugin-dts works for testcase "keep-interface-definition" 1`] = `
"interface Foo {
  a: string;
  b: Array<number>;
}

export { Foo };"
`;

exports[`rollup-plugin-dts works for testcase "remove-function-body" 1`] = `
"function foo()

export { foo };"
`;

exports[`rollup-plugin-dts works for testcase "remove-unexported-function" 1`] = `
"function foo()

export { foo };"
`;

exports[`rollup-plugin-dts works for testcase "remove-unexported-interface" 1`] = `
"interface Foo {}

export { Foo };"
`;
